### PR TITLE
Bug 2115898: Prometheus needs extra permissions (RBAC) for metrics

### PR DESCRIPTION
**Pull request by rauhersu**

### CODE REVIEW

**Comment by rauhersu** [comment]
pinging @jc-rh @donpenney for lgtm

File: config/prometheus/monitor.yaml [hunk 1]
```diff
@@ -18,3 +18,34 @@ spec:
...
+  - pods
+  verbs:
+  - get
+  - list
+  - watch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  creationTimestamp: null
```
Comment (jc-rh) [inline config/prometheus/monitor.yaml:42]: is this needed?

File: bundle/manifests/cluster-group-upgrades-prometheus-metrics-reader_rbac.authorization.k8s.io_v1_role.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  creationTimestamp: null
```
Comment (jc-rh) [inline bundle/manifests/cluster-group-upgrades-prometheus-metrics-reader_rbac.authorization.k8s.io_v1_role.yaml:4]: remove this kind of lines if the files are not generated.

File: config/prometheus/monitor.yaml [hunk 3]
```diff
@@ -18,3 +18,34 @@ spec:
   selector:
     matchLabels:
       control-plane: controller-manager
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  creationTimestamp: null
```
Comment (jc-rh) [inline config/prometheus/monitor.yaml:25]: remove this line and the one in the next object

**Comment by rauhersu** [comment]
pinging @jc-rh for lgtm

**Review by jc-rh — APPROVED** [review]

**Pull request merged**